protostar/starknet/cheaters/contracts.py

```diff
@@ -88,6 +92,44 @@ def apply(self, parent: Self) -> None:
             **self.contract_address_to_class_hash_map,
         }
 
+    async def _transform_calldata_to_cairo_data_by_addr(
+        self,
+        contract_address: Address,
+        function_name: str,
+        calldata: Optional[CairoOrPythonData] = None,
+    ) -> CairoData:
+        contract_address_int = int(contract_address)
+        class_hash = await self.cheatable_state.get_class_hash_at(contract_address_int)
+        return await self._transform_calldata_to_cairo_data(
+            class_hash=int.from_bytes(class_hash, "big"),
+            function_name=function_name,
+            calldata=calldata,
+        )
+
+    async def _transform_calldata_to_cairo_data(
+        self,
+        class_hash: int,
+        function_name: str,
+        calldata: Optional[CairoOrPythonData] = None,
+    ) -> CairoData:
+        if isinstance(calldata, collections.Mapping):
+            contract_class = await self.cheatable_state.get_contract_class(
+                class_hash=int.to_bytes(class_hash, 32, "big")
+            )
+            assert contract_class.abi, f"No abi found for the contract at {class_hash}"
+
+            transformer = from_python_transformer(
+                contract_class.abi, function_name, "inputs"
+            )
+            try:
+                return transformer(calldata)
+            except DataTransformerException as dt_exc:
+                raise ConstructorInputTransformationException(
```

Comment:
Can we test if this error is thrown?

Reply:
[use starkware functions](https://github.com/software-mansion/protostar/pull/1358/commits/7f4ebe10a172d725efcf44aef9e989817bfa8502)

Reply:
[more tests](https://github.com/software-mansion/protostar/pull/1358/commits/77ea28b250c7bb8522e97b69cb4bbd94a060b0f5)